client/go/cmd/clone.go

```diff
@@ -125,20 +120,12 @@ func useCache(cacheFile *os.File) (bool, error) {
 	return stat.Size() > 0 && time.Now().Before(expiry), nil
 }
 
-func getSampleAppsZip() (*os.File, error) {
-	f, err := openOutputFile()
+func fetchSampleAppsZip(cacheDir string, destination string) error {
+	f, err := ioutil.TempFile(cacheDir, "sample-apps")
```

Comment:
Instead of passing `cacheDir` as a separate argument this should just do `filepath.Dir(destination)` so that the temp file directory is always the parent of `destination`.

Reply:
... that was the clean solution I missed. Thanks.